emcc.py

```diff
@@ -2483,6 +2496,8 @@ def consume_arg_file():
         # difference).
         shared.Settings.DEBUG_LEVEL = 3
     elif check_flag('-profiling') or check_flag('--profiling'):
+      if compile_only:
+        exit_with_error('-profiling is a link time setting! (Pass e.g. -g2 at compile stage instead)')
```

Comment:
Same with all of these.  `unused-command-line-argument` seems like a better solution and only breaks those that have `-Werror` enabled.